pxtcompiler/emitter/backjs.ts

```diff
@@ -332,11 +332,16 @@ switch (step) {
                     args.splice(1, 0, procid.mapIdx.toString())
                     write(`  s.retval = ${shimToJs(procid.mapMethod)}(${args.join(", ")});`)
                     write(`  ${frameRef}.fn = doNothing;`)
-                    write(`} else`)
+                    write(`} else {`)
                 }
+                write (`pxsim.check(typeof ${frameRef}.arg0  != "number", "Can't access property of null/undefined.")`)
```

Comment:
Why is `arg0` a number? I guess it's `0`? If so, we should have `check(!!${frameRef}.arg0, ...)` - as we align with JS semantics we should switch to `null` being represented by `null` etc.